lib/gauge_field.cpp

```diff
@@ -375,6 +377,7 @@ namespace quda {
       gParamEx.x[d] += 2 * R[d];
       gParamEx.r[d] = R[d];
     }
+    if (recon != QUDA_RECONSTRUCT_INVALID) gParamEx.reconstruct = recon;
```

Comment:
Should this have an `else errorQuda(...)` path?

Reply:
Yes, probably.  This is code that I have no idea of the origin of, and the lack of an error was likely an oversight.

Reply:
Ah, I remember this code path now, and why it's like that.  It's by design 😄 .  The default value passed to the function is `QUDA_RECONSTRUCT_INVALID`, so we check to see if a different value is passed.  This allows the user to override the reconstruct if desired.  If not, then we don't override, and simply use the reconstruct value from the `in` field.